backend/src/main/java/com/bakdata/conquery/models/config/auth/JwtPkceVerifyingRealmFactory.java

```diff
@@ -230,14 +239,19 @@ private IdpConfiguration retrieveIdpConfiguration(final Client client) {
 		}
 
 
-		final List<JWK> keys = jwks.getKeys();
-		if (keys.size() != 1) {
-			throw new IllegalStateException("Expected exactly 1 jwk for realm but found: " + keys.size());
-		}
+		// Filter for keys that are used for signing (discard encryption keys)
+		final Map<String, PublicKey> signingKeys = jwks.getKeys().stream()
+													   .filter(jwk -> jwk.getPublicKeyUse().equals(JWK.Use.SIG.name()))
```

Comment:
wenn du die operanden swappest kannst du eine potentielle NPE vermeiden ohne ärger

```suggestion
													   .filter(jwk -> JWK.Use.SIG.name().equals(jwk.getPublicKeyUse()))
```